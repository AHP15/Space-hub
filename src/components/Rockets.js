import React from 'react';
import '../styles/Rockets.css';
import Rocket from './Rocket';

const Rockets = () => (
  <section className="rockets">
    <div className="container" />
    <Rocket />
    <Rocket />
    <Rocket />
  </section>
);

export default Rockets;
