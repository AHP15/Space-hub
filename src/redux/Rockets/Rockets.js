const initialState = [];

const rocketsReduser = (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export default rocketsReduser;
