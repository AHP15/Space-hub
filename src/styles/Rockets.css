.rockets {
  padding: 15px;
}

.rockets .card {
  display: flex;
  gap: 15px;
  margin-top: 20px;
}

.rockets .card img {
  width: 250px;
}

.rockets .card button {
  background: #007bff;
  border: none;
  padding: 10px;
  border-radius: 5px;
  cursor: pointer;
  margin-top: 10px;
  color: white;
  font-size: 18px;
}

.rockets .card button.booked {
  background-color: white;
  border: 1px solid #adadad;
  color: #adadad;
}
