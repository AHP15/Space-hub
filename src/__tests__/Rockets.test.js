import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';

import Rockets from '../components/Rockets';
import Store from '../slices/store';

it('testing Rockets', () => {
  const tree = renderer.create(<Provider store={Store}><Rockets /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
