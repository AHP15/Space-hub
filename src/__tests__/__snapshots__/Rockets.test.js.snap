// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Rockets 1`] = `
<section
  className="rockets"
>
  <div
    className="rockets-container"
  >
    <div
      className="card"
    >
      <img
        alt="rocketIMG"
        src="first image"
      />
      <div
        className="rocket-info"
      >
        <h2>
          mission name
        </h2>
        <p>
          mission description
        </p>
        <button
          className="booked"
          onClick={[Function]}
          title="isMember"
          type="button"
        >
          Cancel reservation
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="rocketIMG"
        src="first image"
      />
      <div
        className="rocket-info"
      >
        <h2>
          mission name
        </h2>
        <p>
          mission description
        </p>
        <button
          className=""
          onClick={[Function]}
          title="isMember"
          type="button"
        >
          Reserve rocket
        </button>
      </div>
    </div>
  </div>
</section>
`;
